app.rb

```diff
@@ -0,0 +1,38 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/chitter.rb'
+
+class Chitter_challenge < Sinatra::Base
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  enable :sessions
+
+  get '/' do
+    #@username = Chitter.signup
+    erb :index
+  end
+
+  post '/' do
+    Chitter.signup(username: params[:username], name: params[:name], emailaddress: params[:emailaddress], password: params[:password])
+    @username = params[:username]
+    erb :signup
+  end
+
+  get '/timeline' do
+    @peep = Chitter.timeline
+    time = Time.new
+    @date_posted = time.strftime("%d/%m/%Y")
+    @date_posted2 = time.strftime("%k:%M")
+    
+    erb :timeline
+  end
+  
+  post '/timeline' do
+    Chitter.post(peep: params[:peep])
+    redirect '/timeline'
+  end
+
+  run! if app_file == $0
```

Comment:
controller is nice and "skinny" with no methods over 4-5 lines. Rubocop will cut that statement on line 18 into chunks to make it more readable too.